Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.PageSetupDialog.cs

```diff
@@ -0,0 +1,176 @@
+        // </summary>
+        // 
+        // <returns>
+        // true if the paper section of the dialog box is enabled; otherwise, false. The default is true.
+        // </returns>
+        // [DefaultValue(true)]
+        // public bool AllowPaper {get; set;}
+        
+        // <summary>
+        // Gets or sets a value indicating whether the Printer button is enabled.
+        // </summary>
+        // 
+        // <returns>
+        // true if the Printer button is enabled; otherwise, false. The default is true.
+        // </returns>
+
+        // [DefaultValue(true)]
+        // public bool AllowPrinter {get; set;}
+       
+        // <summary>
+        // Gets or sets a value indicating the <see cref="T:System.Drawing.Printing.PrintDocument"/> to get page settings from.
+        // </summary>
+        // 
+        // <returns>
+        // The <see cref="T:System.Drawing.Printing.PrintDocument"/> to get page settings from. The default is null.
+        // </returns>
+
+        // [DefaultValue(null)]
+        // public PrintDocument Document {get; set;}
+       
+        // <summary>
+        // Gets or sets a value indicating whether the margin settings, when displayed in millimeters, should be automatically converted to and from hundredths of an inch.
+        // </summary>
+        // 
+        // <returns>
+        // true if the margins should be automatically converted; otherwise, false. The default is false.
+        // </returns>
+        // [DefaultValue(false)]
+        // public bool EnableMetric {get; set;}
+        
+        // <summary>
+        // Gets or sets a value indicating the minimum margins, in hundredths of an inch, the user is allowed to select.
+        // </summary>
+        // 
+        // <returns>
+        // The minimum margins, in hundredths of an inch, the user is allowed to select. The default is null.
+        // </returns>
+        public Margins MinMargins
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<Margins>() != null);
+                return default(Margins);
+            }
+            set
+            {
+               
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating the page settings to modify.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The <see cref="T:System.Drawing.Printing.PageSettings"/> to modify. The default is null.
+        /// </returns>
+        /// [DefaultValue(null)]
+        public PageSettings PageSettings {get; set;}
+       
+        // <summary>
+        // Gets or sets the printer settings that are modified when the user clicks the Printer button in the dialog.
+        // </summary>
+        // 
+        // <returns>
+        // The <see cref="T:System.Drawing.Printing.PrinterSettings"/> to modify when the user clicks the Printer button. The default is null.
+        // </returns>
+
+        [DefaultValue(null)]
```

Comment:
It is used in the contract for RunDialog. The value can be null, but when RunDialog is call, it must be non-null.